examples/prom/elliptic_eigenproblem_global_rom.cpp

```diff
@@ -0,0 +1,939 @@
+                   mergeTimer.RealTime());
+        }
+        delete generator;
+        delete options;
+        MPI_Finalize();
+        return 0;
+    }
+
+    // 10. Set up the parallel bilinear forms a(.,.) and m(.,.) on the finite
+    //     element space. The first corresponds to the Laplacian operator -Delta,
+    //     while the second is a simple mass matrix needed on the right hand side
+    //     of the generalized eigenvalue problem below. The boundary conditions
+    //     are implemented by elimination with special values on the diagonal to
+    //     shift the Dirichlet eigenvalues out of the computational range. After
+    //     serial and parallel assembly we extract the corresponding parallel
+    //     matrices A and M.
+    assembleTimer.Start();
+    ConstantCoefficient one(1.0);
+
+    FunctionCoefficient kappa_0(Conductivity);
+    FunctionCoefficient v_0(Potential);
+
+    // Project initial conductivity and potential to visualize initial condition
+    ParGridFunction c_gf(fespace);
+    c_gf.ProjectCoefficient(kappa_0);
+
+    ParGridFunction p_gf(fespace);
+    p_gf.ProjectCoefficient(v_0);
+
+    Array<int> ess_bdr;
+    if (pmesh->bdr_attributes.Size())
+    {
+        ess_bdr.SetSize(pmesh->bdr_attributes.Max());
+        ess_bdr = 1;
+    }
+
+    ParBilinearForm *a = new ParBilinearForm(fespace);
+    a->AddDomainIntegrator(new DiffusionIntegrator(kappa_0));
+    a->AddDomainIntegrator(new MassIntegrator(v_0));
+    a->Assemble();
+    a->EliminateEssentialBCDiag(ess_bdr, 1.0);
+    a->Finalize();
+
+    ParBilinearForm *m = new ParBilinearForm(fespace);
+    m->AddDomainIntegrator(new MassIntegrator(one));
+    m->Assemble();
+    // shift the eigenvalue corresponding to eliminated dofs to a large value
+    m->EliminateEssentialBCDiag(ess_bdr, numeric_limits<double>::min());
+    m->Finalize();
+
+    HypreParMatrix *A = a->ParallelAssemble();
+    HypreParMatrix *M = m->ParallelAssemble();
+
+#if defined(MFEM_USE_SUPERLU) || defined(MFEM_USE_STRUMPACK)
+    Operator * Arow = NULL;
+#endif
+#ifdef MFEM_USE_SUPERLU
+    if (slu_solver)
+    {
+        Arow = new SuperLURowLocMatrix(*A);
+    }
+#endif
+#ifdef MFEM_USE_STRUMPACK
+    if (sp_solver)
+    {
+        Arow = new STRUMPACKRowLocMatrix(*A);
+    }
+#endif
+
+    assembleTimer.Stop();
+
+    delete a;
+    delete m;
+
+    ParGridFunction x(fespace);
+    HypreLOBPCG *lobpcg;
+    Array<double> eigenvalues;
+    DenseMatrix evect;
+
+    // 11. The offline phase
+    if(fom || offline)
+    {
+        // 12. Define and configure the LOBPCG eigensolver and the BoomerAMG
+        //     preconditioner for A to be used within the solver. Set the matrices
+        //     which define the generalized eigenproblem A x = lambda M x.
+        Solver * precond = NULL;
+        if (!slu_solver && !sp_solver && !cpardiso_solver)
+        {
+            HypreBoomerAMG * amg = new HypreBoomerAMG(*A);
```

Comment:
We need to do `delete amg`?

Reply:
I think this should be taken care of already since `amg`,`superlu`, etc are assigned to `precond`, which gets deleted here
https://github.com/LLNL/libROM/blob/92c3c4531a9826460b05be5219749446d38957f8/examples/prom/elliptic_eigenproblem_global_rom.cpp#L491